augur/io/metadata.py

```diff
@@ -133,7 +135,12 @@ def read_metadata_to_dict(metadata, duplicate_reporting=DataErrorMethod.ERROR, i
     with open_file(metadata) as handle:
         # Get sample to determine delimiter
         metadata_sample = handle.read(1024)
-        handle.seek(0)
+
+        if handle.seekable():
+            handle.seek(0)
+        else:
+            metadata_sample_file = StringIO(metadata_sample)
+            handle = chain(metadata_sample_file, handle)
```

Comment:
Very nice technique!

One question: I thought `csv` wanted binary/byte IO not text/string IO?

Reply:
I believe `csv.reader` wants text/string IO based on this line in the [docs](https://docs.python.org/3/library/csv.html#csv.reader):
> If csvfile is a file object, it should be opened with newline=''

binary doesn't support `newline` right?

Reply:
Ah, indeed.